framework/codemodder-base/src/main/java/io/codemodder/CLI.java

```diff
@@ -166,19 +168,17 @@ enum OutputFormat {
   @VisibleForTesting
   static class DefaultFileFinder implements FileFinder {
     @Override
-    public List<Path> findFiles(
-        final List<SourceDirectory> sourceDirectories, final IncludesExcludes includesExcludes) {
+    public List<Path> findFiles(final Path projectDir, final IncludesExcludes includesExcludes) {
       List<Path> allFiles = new ArrayList<>();
-      for (SourceDirectory directory : sourceDirectories) {
-        allFiles.addAll(
-            directory.files().stream()
-                .map(File::new)
-                .filter(includesExcludes::shouldInspect)
-                .map(File::toPath)
-                .filter(
-                    f -> !Files.isSymbolicLink(f)) // could cause infinite loop if we follow links
-                .sorted()
-                .toList());
+      try (Stream<Path> paths = Files.walk(projectDir)) {
+        paths
+            .filter(Files::isRegularFile)
+            .filter(p -> !Files.isSymbolicLink(p)) // could cause infinite loop if we follow links
+            .filter(p -> includesExcludes.shouldInspect(p.toFile()))
+            .sorted()
+            .forEach(allFiles::add);
```

Comment:
```suggestion
      final List<Path> allFiles;
      try (Stream<Path> paths = Files.walk(projectDir)) {
        allFiles = paths
            .filter(Files::isRegularFile)
            .filter(p -> !Files.isSymbolicLink(p)) // could cause infinite loop if we follow links
            .filter(p -> includesExcludes.shouldInspect(p.toFile()))
            .sorted()
            .toList();
```

Lets use `Stream.toList()` inside the try-with-resources rather than creating a list and adding to it inside a `forEach`

Reply:
chatgpt prompts to get this, because I looked at it and thought it was a bit odd:

> how to use Files.walk java api properly  

outputted something very similar to what you have here

> How can I see Files.walk and the Stream Collect API

Gave what i suggested